lib/crystalball/map_generator/simple_map.rb

```diff
@@ -4,30 +4,57 @@ module Crystalball
   class MapGenerator
     # Basic map object for storing execution maps to storage
     class SimpleMap
-      def initialize(storage)
-        @storage = storage
-        @raw_map = {}
+      extend Forwardable
+
+      # Simple data object for map metadata
+      class Metadata
+        attr_accessor :commit, :type
+
+        def initialize(commit: nil, type: nil)
+          @commit = commit
+          @type = type
+        end
+
+        def to_h
+          {type: type, commit: commit}
+        end
       end
 
-      def load
-        self.raw_map = storage.load
+      attr_reader :cases, :metadata
+
+      delegate %i[commit commit=] => :metadata
+
+      def initialize(storage, metadata: nil, cases: {})
+        @storage = storage
+        @cases = cases
+        @metadata = Metadata.new(metadata || {type: self.class.name})
```

Comment:
I think it should be `Metadata.new(type: self.class.name, **metadata)`, so type can be overriden, but it still should always be present, no?

Reply:
Actually do we really need metadata object? It's 2 fields only and complicates things.

Reply:
Well its a nice interface where to store your non-case related information. If user needs to store something else they can easily extend their own Metadata class.